aptos-move/framework/supra-framework/sources/supra_governance.move

```diff
@@ -581,10 +798,77 @@ module supra_framework::supra_governance {
         }
     }
 
+    /// Vote on proposal with `proposal_id` and all voting power from `stake_pool`.
+    public entry fun supra_vote(
+        voter: &signer,
+        proposal_id: u64,
+        should_pass: bool,
+    ) acquires SupraApprovedExecutionHashes, SupraGovernanceEvents, SupraGovernanceConfig {
+        supra_vote_internal(voter, proposal_id, should_pass);
+    }
+
+    /// Vote on proposal with `proposal_id` and specified voting_power from `stake_pool`.
+    /// If voting_power is more than all the left voting power of `stake_pool`, use all the left voting power.
+    /// If a stake pool has already voted on a proposal before partial governance voting is enabled, the stake pool
+    /// cannot vote on the proposal even after partial governance voting is enabled.
+    fun supra_vote_internal(
+        voter: &signer,
+        proposal_id: u64,
+        should_pass: bool,
+    ) acquires SupraApprovedExecutionHashes, SupraGovernanceEvents, SupraGovernanceConfig {
+        let voter_address = signer::address_of(voter);
+
+        let supra_governance_config = borrow_global<SupraGovernanceConfig>(@supra_framework);
+        assert!(vector::contains(&supra_governance_config.voters, &signer::address_of(voter)), error::permission_denied(EACCOUNT_NOT_AUTHORIZED));
+
+        // The voter's stake needs to be locked up at least as long as the proposal's expiration.
+        let proposal_expiration = multisig_voting::get_proposal_expiration_secs<GovernanceProposal>(
+            @supra_framework,
+            proposal_id
+        );
+        assert!(timestamp::now_seconds() >= proposal_expiration, error::invalid_argument(EPROPOSAL_IS_EXPIRE));
+
+        multisig_voting::vote<GovernanceProposal>(
+            &governance_proposal::create_empty_proposal(),
```

Comment:
Did you check what this `create_empty_proposal` does?

Reply:
This function simply return `GovernanceProposal {}`